*: move Notary contract out of P2PSigExtensions under Echidna hardfork

AnnaShaleva opened the pull request:
Close #3464.

AnnaShaleva commented:
Moved Notary functionality from Domovoi under Echidna hardfork.

AnnaShaleva commented:
Oh-oh, probably need to move it under next hardfork. Ref. https://github.com/nspcc-dev/neo-go/pull/3554 and https://github.com/neo-project/neo/pull/3454.

roman-khimov commented:
It still can be a part of Echidna. But for now we don't need to change anything anyway, it's totally C#-dependent.

roman-khimov commented on pkg/core/native/notary.go:
```diff
@@ -52,7 +52,10 @@ const (
-var maxNotValidBeforeDeltaKey = []byte{10}
+var (
+	maxNotValidBeforeDeltaKey = []byte{10}
+	activeIn                  = config.HFEchidna
```
Don't forget it's package-level, better move into `ActiveIn()`.

AnnaShaleva replied:
Sure, fixed.

roman-khimov commented on pkg/core/native/notary.go:
```diff
@@ -97,17 +100,17 @@ func newNotary() *Notary {
 
 	desc = newDescriptor("verify", smartcontract.BoolType,
-		manifest.NewParameter("signature", smartcontract.SignatureType))
+		manifest.NewParameter("signature", smartcontract.ByteArrayType))
```
Meh. Bad one, but OK.

AnnaShaleva replied:
C# node doesn't support SC types in native method wrappers out of the box, and I wasn't able to properly add this support.

AnnaShaleva commented:
@roman-khimov, I can't request review from copilot, do I need some extra rights for that?

roman-khimov commented:
> --- FAIL: TestDBRestoreDump (0.10s)
    --- PASS: TestDBRestoreDump/excessive_restore_parameters (0.00s)
panic: item with id = -10 and key = 0a is not initialized [recovered]
	panic: item with id = -10 and key = 0a is not initialized

AnnaShaleva commented:
>--- FAIL: TestDBRestoreDump (0.10s)
--- PASS: TestDBRestoreDump/excessive_restore_parameters (0.00s)
panic: item with id = -10 and key = 0a is not initialized [recovered]
panic: item with id = -10 and key = 0a is not initialized

It's due to https://github.com/nspcc-dev/neo-go/pull/3478#discussion_r2068470763. Unexpected. I reverted it and renamed `activeIn` to `notaryActiveIn`, let it be the global variable.
